keps/sig-node/3695-pod-resources-for-dra/README.md

```diff
@@ -0,0 +1,376 @@
+    repeated string device_ids = 2;
+    TopologyInfo topology = 3;
+}
+
+// Topology describes hardware topology of the resource
+message TopologyInfo {
+    repeated NUMANode nodes = 1;
+}
+
+// NUMA representation of NUMA node
+message NUMANode {
+    int64 ID = 1;
+}
+
+// DynamicResource contains information about the devices assigned to a container by DRA
+message DynamicResource {
+    string class_name = 1;
+    string claim_name = 2;
+    repeated string cdi_devices = 3;
+}
+
+// GetPodResourcesRequest contains information about the pod
+message GetPodResourcesRequest {
+    string name = 1;
+    string namespace = 2;
+}
+
+// GetPodResourcesResponse contains information about the pod the devices
+message GetPodResourcesResponse {
+    PodResources pod_resources = 1;
+}
+
+```
+
+
+##### Unit tests
+
+- `k8s.io/kubernetes/pkg/kubelet/apis/podresources`: `01-24-2023` - `61.5%`
+
+##### Integration tests
+
+ These cases will be added in the existing integration tests:
+  - Feature gate enable/disable tests.
+  - Get API work with DRA and device plugin.
+  - List API work with DRA and Device plugin.
+
+##### e2e tests
+
+These cases will be added in the existing e2e tests:
+  - Feature gate enable/disable tests.
+  - Get API work with DRA and device plugin.
+  - List API work with DRA and Device plugin.
+
+### Graduation Criteria
+
+#### Alpha
+
+- [ ] Feature implemented behind a feature flag.
+- [ ] e2e tests completed and enabled.
+
+#### Beta
+
+- [ ] Gather feedback from consumers of the DRA feature and k8snetworkplumbingwg working group
+- [ ] No major bugs reported in the previous cycle.
+
+#### GA
+
+- [ ] Allowing time for feedback (1 year).
+- [ ] Risks have been addressed.
+
+### Upgrade / Downgrade Strategy
+
+With gRPC the version is part of the service name.
+Old versions and new versions should always be served and listened by the kubelet.
+
+To a cluster admin upgrading to the newest API version, means upgrading Kubernetes to a newer version as well as upgrading the monitoring component.
+
+To a vendor changes in the API should always be backwards compatible.
+
+### Version Skew Strategy
+
+Kubelet will always be backwards compatible, so going forward existing plugins are not expected to break.
+
+
+## Production Readiness Review Questionnaire
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name for retrieving resources allocated by DRA: `DynamicResourceAllocation` and `PodResourcesDynamicResources`.
+  - Feature gate name for Get method: `PodResourcesGet`. In case `DynamicResourceAllocation` or the `PodResourcesDynamicResources`
+    are disabled and `PodResourcesGet` is enabled the Get will retrieve resources allocated by device plugin, memory and cpus.
+    In case `PodResourcesGet`, `DynamicResourceAllocation` and `PodResourcesDynamicResources` are enabled the Get method will retrieve
```

Comment:
```suggestion
    In case `PodResourcesGet`, `DynamicResourceAllocation` and `PodResourcesDynamicResources` are all enabled, the Get method will
```